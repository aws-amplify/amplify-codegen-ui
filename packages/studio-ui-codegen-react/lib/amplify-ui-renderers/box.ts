import { ViewProps as BoxProps } from '@aws-amplify/ui-react';

import { StudioComponentChild } from '@amzn/amplify-ui-codegen-schema';

import { factory, JsxChild, JsxElement } from 'typescript';
import { ReactComponentWithChildrenRenderer } from '../react-component-with-children-renderer';

export default class BoxRenderer extends ReactComponentWithChildrenRenderer<BoxProps> {
  renderElement(renderChildren: (children: StudioComponentChild[]) => JsxChild[]): JsxElement {
    const tagName = 'View';

    const childrenJsx = this.component.children ? renderChildren(this.component.children) : [];
    const element = factory.createJsxElement(
      this.renderOpeningElement(tagName),
      childrenJsx,
      factory.createJsxClosingElement(factory.createIdentifier(tagName)),
    );

    this.importCollection.addImport('@aws-amplify/ui-react', tagName);

    return element;
  }
}
