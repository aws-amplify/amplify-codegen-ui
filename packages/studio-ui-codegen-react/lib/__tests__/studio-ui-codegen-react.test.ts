import { StudioComponent } from '@amzn/amplify-ui-codegen-schema';
import { StudioTemplateRendererFactory } from '@amzn/studio-ui-codegen';
import fs from 'fs';
import { join } from 'path';
import { ModuleKind, ScriptTarget, ScriptKind, ReactRenderConfig } from '..';
import { AmplifyRenderer } from '../amplify-ui-renderers/amplify-renderer';

function loadSchemaFromJSONFile(jsonSchemaFile: string): StudioComponent {
  return JSON.parse(
    fs.readFileSync(join(__dirname, 'studio-ui-json', `${jsonSchemaFile}.json`), 'utf-8'),
  ) as StudioComponent;
}

function generateWithAmplifyRenderer(
  jsonSchemaFile: string,
  renderConfig: ReactRenderConfig = {},
  isSampleCodeSnippet = false,
): string {
  const schema = loadSchemaFromJSONFile(jsonSchemaFile);
  const rendererFactory = new StudioTemplateRendererFactory(
    (component: StudioComponent) => new AmplifyRenderer(component, renderConfig),
  );
  if (isSampleCodeSnippet) {
    return rendererFactory.buildRenderer(schema).renderSampleCodeSnippet().compText;
  }
  return rendererFactory.buildRenderer(schema).renderComponent().componentText;
}

describe('amplify render tests', () => {
  describe('basic component tests', () => {
    it('should generate a simple box component', () => {
      const generatedCode = generateWithAmplifyRenderer('boxTest');
      expect(generatedCode).toMatchSnapshot();
    });

    it('should generate a simple button component', () => {
      const generatedCode = generateWithAmplifyRenderer('buttonGolden');
      expect(generatedCode).toMatchSnapshot();
    });

    it('should generate a simple text component', () => {
      const generatedCode = generateWithAmplifyRenderer('textGolden');
      expect(generatedCode).toMatchSnapshot();
    });

    it('should generate a simple badge component', () => {});

    it('should generate a simple card component', () => {});

    it('should generate a simple divider component', () => {});

    it('should generate a simple flex component', () => {});

    it('should generate a simple image component', () => {});

    it('should generate a simple string component', () => {});
  });

  describe('complex component tests', () => {
    it('should generate a button within a box component', () => {
      const generatedCode = generateWithAmplifyRenderer('boxGolden');
      expect(generatedCode).toMatchSnapshot();
    });

    it('should generate a component with custom child', () => {
      const generatedCode = generateWithAmplifyRenderer('customChild');
      expect(generatedCode).toMatchSnapshot();
    });

    it('should generate a component with exposeAs prop', () => {
      const generatedCode = generateWithAmplifyRenderer('exposedAsTest');
      expect(generatedCode).toMatchSnapshot();
    });
  });

  describe('sample code snippet tests', () => {
    it('should generate a sample code snippet for components', () => {
      const generatedCode = generateWithAmplifyRenderer('sampleCodeSnippet');
      expect(generatedCode).toMatchSnapshot();
    });
  });

  describe('component with data binding', () => {
    it('should add model imports', () => {
      const generatedCode = generateWithAmplifyRenderer('componentWithDataBinding');
      expect(generatedCode).toMatchSnapshot();
    });

    it('should render collection data binding', () => {
      const generatedCode = generateWithAmplifyRenderer('collectionWithBinding');
      expect(generatedCode).toMatchSnapshot();
    });
  });

  describe('custom render config', () => {
    it('should render ES5', () => {
      expect(
        generateWithAmplifyRenderer('boxGolden', { target: ScriptTarget.ES5, script: ScriptKind.JS }),
      ).toMatchSnapshot();
    });

    it('should render JSX', () => {
      expect(generateWithAmplifyRenderer('boxGolden', { script: ScriptKind.JSX })).toMatchSnapshot();
    });

    it('should render common JS', () => {
      expect(
        generateWithAmplifyRenderer('boxGolden', { module: ModuleKind.CommonJS, script: ScriptKind.JS }),
      ).toMatchSnapshot();
    });
  });
});
