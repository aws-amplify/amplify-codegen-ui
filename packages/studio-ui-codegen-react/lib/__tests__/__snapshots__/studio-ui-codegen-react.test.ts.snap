// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`amplify render tests basic component tests should generate a simple box component 1`] = `
"/* eslint-disable */
import React from \\"react\\";
import {
  EscapeHatchProps,
  View,
  getOverrideProps,
} from \\"@aws-amplify/ui-react\\";

export type TestProps = {
  overrides?: EscapeHatchProps | undefined | null,
};
export default function Test(props: TestProps): JSX.Element {
  const overrides = { ...props.overrides };
  return (
    <View
      fontFamily=\\"Times New Roman\\"
      fontSize=\\"20px\\"
      {...props}
      {...getOverrideProps(overrides, \\"Box\\")}
    ></View>
  );
}
"
`;

exports[`amplify render tests basic component tests should generate a simple button component 1`] = `
"/* eslint-disable */
import React from \\"react\\";
import {
  Button,
  EscapeHatchProps,
  getOverrideProps,
} from \\"@aws-amplify/ui-react\\";

export type CustomButtonProps = {
  overrides?: EscapeHatchProps | undefined | null,
};
export default function CustomButton(props: CustomButtonProps): JSX.Element {
  const overrides = { ...props.overrides };
  return (
    <Button
      color=\\"#ff0000\\"
      width=\\"20\\"
      {...props}
      {...getOverrideProps(overrides, \\"Button\\")}
    ></Button>
  );
}
"
`;

exports[`amplify render tests basic component tests should generate a simple text component 1`] = `
"/* eslint-disable */
import React from \\"react\\";
import {
  EscapeHatchProps,
  Text,
  getOverrideProps,
} from \\"@aws-amplify/ui-react\\";

export type CustomTextProps = {
  overrides?: EscapeHatchProps | undefined | null,
};
export default function CustomText(props: CustomTextProps): JSX.Element {
  const overrides = { ...props.overrides };
  return (
    <Text
      color=\\"#ff0000\\"
      width=\\"20px\\"
      {...props}
      {...getOverrideProps(overrides, \\"Text\\")}
    >
      Text Value
    </Text>
  );
}
"
`;

exports[`amplify render tests collection should render collection with data binding 1`] = `
"/* eslint-disable */
import React from \\"react\\";
import { User, UserPreferences } from \\"../models\\";
import {
  Button,
  Collection,
  EscapeHatchProps,
  Flex,
  getOverrideProps,
} from \\"@aws-amplify/ui-react\\";

export type CollectionOfCustomButtonsProps = {
  width?: Number,
  backgroundColor?: String,
  buttonColor?: UserPreferences,
  items?: any[],
} & {
  overrides?: EscapeHatchProps | undefined | null,
};
export default function CollectionOfCustomButtons(
  props: CollectionOfCustomButtonsProps
): JSX.Element {
  const { width, backgroundColor, buttonColor, items } = props;
  const overrides = { ...props.overrides };
  const buttonUserFilter = {
    and: [
      { field: \\"age\\", operand: \\"10\\", operator: \\"gt\\" },
      { field: \\"lastName\\", operand: \\"L\\", operator: \\"beginsWith\\" },
    ],
  };
  const buttonUser =
    items !== undefined
      ? items
      : useDataStoreBinding({
          type: \\"collection\\",
          model: User,
          criteria: buttonUserFilter,
        }).items;
  const buttonColorFilter = {
    field: \\"userID\\",
    operand: \\"user@email.com\\",
    operator: \\"eq\\",
  };
  const { item: buttonColor } = useDataStoreBinding({
    type: \\"record\\",
    model: UserPreferences,
    criteria: buttonColorFilter,
  });
  return (
    <Collection
      type=\\"list\\"
      isPaginated={true}
      gap=\\"1.5rem\\"
      backgroundColor={backgroundColor}
      items={buttonUser}
      {...props}
      {...getOverrideProps(overrides, \\"Collection\\")}
    >
      {(item, index) => (
        <Flex {...getOverrideProps(overrides, \\"Collection.Flex\\")}>
          <Button
            label={item.username || \\"hspain@gmail.com\\"}
            labelWidth={width}
            backgroundColor={buttonColor.favoriteColor}
            disabled={isDisabled}
            {...getOverrideProps(overrides, \\"Collection.Flex.Button\\")}
          ></Button>
        </Flex>
      )}
    </Collection>
  );
}
"
`;

exports[`amplify render tests collection should render collection with data binding with no predicate 1`] = `
"/* eslint-disable */
import React from \\"react\\";
import {
  Collection,
  EscapeHatchProps,
  ListingCard,
  findChildOverrides,
  getOverrideProps,
} from \\"@aws-amplify/ui-react\\";
import { UntitledModel } from \\"../models\\";

export type ListingCardCollectionProps = {
  items?: any[],
} & {
  overrides?: EscapeHatchProps | undefined | null,
};
export default function ListingCardCollection(
  props: ListingCardCollectionProps
): JSX.Element {
  const { items } = props;
  const overrides = { ...props.overrides };
  const bananas =
    items !== undefined
      ? items
      : useDataStoreBinding({
          type: \\"collection\\",
          model: UntitledModel,
        }).items;
  return (
    <Collection
      isPaginated=\\"true\\"
      collectionType=\\"grid\\"
      type=\\"list\\"
      columns=\\"2\\"
      order=\\"left-to-right\\"
      items={bananas}
      {...props}
      {...getOverrideProps(overrides, \\"Collection\\")}
    >
      {(item, index) => (
        <ListingCard
          marginRight=\\"0\\"
          marginBottom=\\"0\\"
          marginTop=\\"0\\"
          marginLeft=\\"0\\"
          {...findChildOverrides(props.overrides, \\"ListingCard\\")}
        ></ListingCard>
      )}
    </Collection>
  );
}
"
`;

exports[`amplify render tests collection should render collection without data binding 1`] = `
"/* eslint-disable */
import React from \\"react\\";
import {
  Collection,
  EscapeHatchProps,
  ListingCard,
  findChildOverrides,
  getOverrideProps,
} from \\"@aws-amplify/ui-react\\";

export type ListingCardCollectionProps = {
  items?: any[],
} & {
  overrides?: EscapeHatchProps | undefined | null,
};
export default function ListingCardCollection(
  props: ListingCardCollectionProps
): JSX.Element {
  const { items } = props;
  const overrides = { ...props.overrides };
  return (
    <Collection
      type=\\"list\\"
      isPaginated=\\"true\\"
      items={items}
      {...props}
      {...getOverrideProps(overrides, \\"Collection\\")}
    >
      {(item, index) => (
        <ListingCard
          {...findChildOverrides(props.overrides, \\"ListingCard\\")}
        ></ListingCard>
      )}
    </Collection>
  );
}
"
`;

exports[`amplify render tests collection should render collection with data binding and sort 1`] = `
"/* eslint-disable */
import React from \\"react\\";
import { User, UserPreferences } from \\"../models\\";
import {
  Button,
  Collection,
  EscapeHatchProps,
  Flex,
  convertSortPredicatesToDataStore,
  getOverrideProps,
} from \\"@aws-amplify/ui-react\\";

export type CollectionOfCustomButtonsProps = {
  width?: Number,
  backgroundColor?: String,
  buttonColor?: UserPreferences,
  items?: any[],
} & {
  overrides?: EscapeHatchProps | undefined | null,
};
export default function CollectionOfCustomButtons(
  props: CollectionOfCustomButtonsProps
): JSX.Element {
  const { width, backgroundColor, buttonColor, items } = props;
  const overrides = { ...props.overrides };
  const buttonUserFilter = {
    and: [
      { field: \\"age\\", operand: \\"10\\", operator: \\"gt\\" },
      { field: \\"lastName\\", operand: \\"L\\", operator: \\"beginsWith\\" },
    ],
  };
  const buttonUserSort = convertSortPredicatesToDataStore([
    { field: \\"firstName\\", direction: \\"ASC\\" },
    { field: \\"lastName\\", direction: \\"DESC\\" },
  ]);
  const buttonUserPagination = {
    sort: buttonUserSort,
  };
  const buttonUser =
    items !== undefined
      ? items
      : useDataStoreBinding({
          type: \\"collection\\",
          model: User,
          criteria: buttonUserFilter,
          pagination: buttonUserPagination,
        }).items;
  const buttonColorFilter = {
    field: \\"userID\\",
    operand: \\"user@email.com\\",
    operator: \\"eq\\",
  };
  const { item: buttonColor } = useDataStoreBinding({
    type: \\"record\\",
    model: UserPreferences,
    criteria: buttonColorFilter,
  });
  return (
    <Collection
      type=\\"list\\"
      isPaginated={true}
      gap=\\"1.5rem\\"
      backgroundColor={backgroundColor}
      items={buttonUser}
      {...props}
      {...getOverrideProps(overrides, \\"Collection\\")}
    >
      {(item, index) => (
        <Flex {...getOverrideProps(overrides, \\"Collection.Flex\\")}>
          <Button
            label={item.username || \\"hspain@gmail.com\\"}
            labelWidth={width}
            backgroundColor={buttonColor.favoriteColor}
            disabled={isDisabled}
            {...getOverrideProps(overrides, \\"Collection.Flex.Button\\")}
          ></Button>
        </Flex>
      )}
    </Collection>
  );
}
"
`;

exports[`amplify render tests complex component tests should generate a button within a box component 1`] = `
"/* eslint-disable */
import React from \\"react\\";
import {
  Button,
  EscapeHatchProps,
  View,
  getOverrideProps,
} from \\"@aws-amplify/ui-react\\";

export type BoxWithButtonProps = {
  overrides?: EscapeHatchProps | undefined | null,
};
export default function BoxWithButton(props: BoxWithButtonProps): JSX.Element {
  const overrides = { ...props.overrides };
  return (
    <View {...props} {...getOverrideProps(overrides, \\"Box\\")}>
      <Button
        color=\\"#ff0000\\"
        width=\\"20px\\"
        {...getOverrideProps(overrides, \\"Box.Button\\")}
      ></Button>
    </View>
  );
}
"
`;

exports[`amplify render tests complex component tests should generate a component with custom child 1`] = `
"/* eslint-disable */
import React from \\"react\\";
import {
  CustomButton,
  EscapeHatchProps,
  View,
  findChildOverrides,
  getOverrideProps,
} from \\"@aws-amplify/ui-react\\";

export type BoxWithCustomButtonProps = {
  overrides?: EscapeHatchProps | undefined | null,
};
export default function BoxWithCustomButton(
  props: BoxWithCustomButtonProps
): JSX.Element {
  const overrides = { ...props.overrides };
  return (
    <View {...props} {...getOverrideProps(overrides, \\"Box\\")}>
      <CustomButton
        color=\\"#ff0000\\"
        width=\\"20px\\"
        {...findChildOverrides(props.overrides, \\"CustomButton\\")}
      ></CustomButton>
    </View>
  );
}
"
`;

exports[`amplify render tests complex component tests should generate a component with exposeAs prop 1`] = `
"/* eslint-disable */
import React from \\"react\\";
import {
  Button,
  EscapeHatchProps,
  View,
  getOverrideProps,
} from \\"@aws-amplify/ui-react\\";

export type BoxWithButtonProps = {
  overrides?: EscapeHatchProps | undefined | null,
};
export default function BoxWithButton(props: BoxWithButtonProps): JSX.Element {
  const overrides = { ...props.overrides };
  return (
    <View {...props} {...getOverrideProps(overrides, \\"Box\\")}>
      <Button
        color=\\"#ff0000\\"
        width=\\"20px\\"
        {...getOverrideProps(overrides, \\"Box.Button\\")}
      ></Button>
    </View>
  );
}
"
`;

exports[`amplify render tests component with binding should render build property on Text 1`] = `
"/* eslint-disable */
import React from \\"react\\";
import {
  EscapeHatchProps,
  Text,
  getOverrideProps,
} from \\"@aws-amplify/ui-react\\";

export type TextWithDataBindingProps = {
  textValue?: String,
} & {
  overrides?: EscapeHatchProps | undefined | null,
};
export default function TextWithDataBinding(
  props: TextWithDataBindingProps
): JSX.Element {
  const { textValue } = props;
  const overrides = { ...props.overrides };
  return (
    <Text
      color=\\"#ff0000\\"
      width=\\"20px\\"
      {...props}
      {...getOverrideProps(overrides, \\"Text\\")}
    >
      {textValue}
    </Text>
  );
}
"
`;

exports[`amplify render tests component with variants should render variants with options provided 1`] = `
"/* eslint-disable */
import React from \\"react\\";
import {
  Button,
  EscapeHatchProps,
  getOverrideProps,
  getOverridesFromVariants,
} from \\"@aws-amplify/ui-react\\";

export type CustomButtonProps = {
  variant: \\"primary\\" | \\"secondary\\",
  size?: \\"large\\",
} & {
  overrides?: EscapeHatchProps | undefined | null,
};
export default function CustomButton(props: CustomButtonProps): JSX.Element {
  const {} = props;
  const variants = [
    {
      variantValues: {
        variant: \\"primary\\",
      },
      overrides: {
        Button: {
          fontSize: \\"12px\\",
        },
      },
    },
    {
      variantValues: {
        variant: \\"secondary\\",
      },
      overrides: {
        Button: {
          fontSize: \\"40px\\",
        },
      },
    },
    {
      variantValues: {
        variant: \\"primary\\",
        size: \\"large\\",
      },
      overrides: {
        Button: {
          width: \\"500\\",
        },
      },
    },
  ];
  const overrides = {
    ...getOverridesFromVariants(variants, props),
    ...props.overrides,
  };
  return (
    <Button {...props} {...getOverrideProps(overrides, \\"Button\\")}></Button>
  );
}
"
`;

exports[`amplify render tests component with data binding should add model imports 1`] = `
"/* eslint-disable */
import React from \\"react\\";
import { User } from \\"../models\\";
import {
  Button,
  EscapeHatchProps,
  getOverrideProps,
} from \\"@aws-amplify/ui-react\\";

export type ComponentWithDataBindingProps = {
  width?: Number,
  isDisabled?: Boolean,
  buttonUser?: User,
  buttonColor?: String,
} & {
  overrides?: EscapeHatchProps | undefined | null,
};
export default function ComponentWithDataBinding(
  props: ComponentWithDataBindingProps
): JSX.Element {
  const { width, isDisabled, buttonUser, buttonColor } = props;
  const overrides = { ...props.overrides };
  return (
    <Button
      label={buttonUser.username || \\"hspain@gmail.com\\"}
      labelWidth={width}
      disabled={isDisabled}
      {...props}
      {...getOverrideProps(overrides, \\"Button\\")}
    ></Button>
  );
}
"
`;

exports[`amplify render tests component with data binding should not have useDataStoreBinding when there is no predicate 1`] = `
"/* eslint-disable */
import React from \\"react\\";
import { UntitledModel } from \\"../models\\";
import {
  EscapeHatchProps,
  Flex,
  Text,
  getOverrideProps,
} from \\"@aws-amplify/ui-react\\";

export type SectionHeadingProps = {
  newProp6fd1?: UntitledModel,
} & {
  overrides?: EscapeHatchProps | undefined | null,
};
export default function SectionHeading(
  props: SectionHeadingProps
): JSX.Element {
  const { newProp6fd1 } = props;
  const overrides = { ...props.overrides };
  return (
    <Flex
      padding=\\"0px 0px 10px 0px\\"
      backgroundColor=\\"rgb(255,255,255)\\"
      direction=\\"column\\"
      {...props}
      {...getOverrideProps(overrides, \\"Flex\\")}
    >
      <Flex
        tes={newProp6fd1.untitledfield}
        {...getOverrideProps(overrides, \\"Flex.Flex\\")}
      >
        <Text
          fontFamily=\\"Inter\\"
          color=\\"rgb(15.539060980081558,29.73937589675188,47.8125)\\"
          textAlign=\\"left\\"
          fontSize=\\"48px\\"
          lineHeight=\\"48px\\"
          fontWeight=\\"300\\"
          {...getOverrideProps(overrides, \\"Flex.Flex.Text\\")}
        >
          Heading 2
        </Text>
      </Flex>
      <Text
        fontFamily=\\"Inter\\"
        color=\\"rgb(0,0,0)\\"
        textAlign=\\"left\\"
        fontSize=\\"32px\\"
        lineHeight=\\"48px\\"
        fontWeight=\\"400\\"
        {...getOverrideProps(overrides, \\"Flex.Text\\")}
      >
        subtitle
      </Text>
    </Flex>
  );
}
"
`;

exports[`amplify render tests concat and conditional transform should render component with concatenation prop 1`] = `
"/* eslint-disable */
import React from \\"react\\";
import { User } from \\"../models\\";
import {
  Button,
  EscapeHatchProps,
  getOverrideProps,
} from \\"@aws-amplify/ui-react\\";

export type CustomButtonProps = {
  width?: Number,
  buttonUser?: User,
  buttonColor?: String,
} & {
  overrides?: EscapeHatchProps | undefined | null,
};
export default function CustomButton(props: CustomButtonProps): JSX.Element {
  const { width, buttonUser, buttonColor } = props;
  const overrides = { ...props.overrides };
  return (
    <Button
      label={\`\${buttonUser.firstname || \\"Harrison\\"}\${\\" \\"}\${
        buttonUser.lastname || \\"Spain\\"
      }\`}
      labelWidth={width}
      {...props}
      {...getOverrideProps(overrides, \\"Button\\")}
    ></Button>
  );
}
"
`;

exports[`amplify render tests concat and conditional transform should render component with conditional prop 1`] = `
"/* eslint-disable */
import React from \\"react\\";
import { User } from \\"../models\\";
import {
  Button,
  EscapeHatchProps,
  getOverrideProps,
} from \\"@aws-amplify/ui-react\\";

export type CustomButtonProps = {
  width?: Number,
  buttonUser?: User,
  buttonColor?: String,
} & {
  overrides?: EscapeHatchProps | undefined | null,
};
export default function CustomButton(props: CustomButtonProps): JSX.Element {
  const { width, buttonUser, buttonColor } = props;
  const overrides = { ...props.overrides };
  return (
    <Button
      label={\`\${buttonUser.firstname || \\"Harrison\\"}\${\\" \\"}\${
        buttonUser.lastname || \\"Spain\\"
      }\`}
      labelWidth={width}
      disabled={buttonUser.isLoggedIn === true ? true : false}
      prompt={
        buttonUser.age > 18
          ? \`\${buttonUser.firstname}\${\\", cast your vote.\\"}\`
          : \\"Sorry you cannot vote\\"
      }
      backgroundColor={
        buttonUser.isLoggedIn === true
          ? buttonUser.loggedInColor
          : buttonUser.loggedOutColor
      }
      {...props}
      {...getOverrideProps(overrides, \\"Button\\")}
    ></Button>
  );
}
"
`;

exports[`amplify render tests custom render config should render ES5 1`] = `
"var __assign =
  (this && this.__assign) ||
  function () {
    __assign =
      Object.assign ||
      function (t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
          s = arguments[i];
          for (var p in s)
            if (Object.prototype.hasOwnProperty.call(s, p)) t[p] = s[p];
        }
        return t;
      };
    return __assign.apply(this, arguments);
  };
/* eslint-disable */
import React from \\"react\\";
import { Button, View, getOverrideProps } from \\"@aws-amplify/ui-react\\";
export default function BoxWithButton(props) {
  var overrides = __assign({}, props.overrides);
  return React.createElement(
    View,
    __assign({}, props, getOverrideProps(overrides, \\"Box\\")),
    React.createElement(
      Button,
      __assign(
        { color: \\"#ff0000\\", width: \\"20px\\" },
        getOverrideProps(overrides, \\"Box.Button\\")
      )
    )
  );
}
"
`;

exports[`amplify render tests custom render config should render JSX 1`] = `
"/* eslint-disable */
import React from \\"react\\";
import { Button, View, getOverrideProps } from \\"@aws-amplify/ui-react\\";
export default function BoxWithButton(props) {
  const overrides = Object.assign({}, props.overrides);
  return (
    <View {...props} {...getOverrideProps(overrides, \\"Box\\")}>
      <Button
        color=\\"#ff0000\\"
        width=\\"20px\\"
        {...getOverrideProps(overrides, \\"Box.Button\\")}
      ></Button>
    </View>
  );
}
"
`;

exports[`amplify render tests custom render config should render common JS 1`] = `
"\\"use strict\\";
var __importDefault =
  (this && this.__importDefault) ||
  function (mod) {
    return mod && mod.__esModule ? mod : { default: mod };
  };
Object.defineProperty(exports, \\"__esModule\\", { value: true });
/* eslint-disable */
const react_1 = __importDefault(require(\\"react\\"));
const ui_react_1 = require(\\"@aws-amplify/ui-react\\");
function BoxWithButton(props) {
  const overrides = Object.assign({}, props.overrides);
  return react_1.default.createElement(
    ui_react_1.View,
    Object.assign({}, props, ui_react_1.getOverrideProps(overrides, \\"Box\\")),
    react_1.default.createElement(
      ui_react_1.Button,
      Object.assign(
        { color: \\"#ff0000\\", width: \\"20px\\" },
        ui_react_1.getOverrideProps(overrides, \\"Box.Button\\")
      )
    )
  );
}
exports.default = BoxWithButton;
"
`;

exports[`amplify render tests sample code snippet tests should generate a sample code snippet for components 1`] = `
"/* eslint-disable */
import React from \\"react\\";
import {
  CustomButton,
  EscapeHatchProps,
  View,
  findChildOverrides,
  getOverrideProps,
} from \\"@aws-amplify/ui-react\\";

export type BoxWithButtonProps = {
  overrides?: EscapeHatchProps | undefined | null,
};
export default function BoxWithButton(props: BoxWithButtonProps): JSX.Element {
  const overrides = { ...props.overrides };
  return (
    <View padding-left {...props} {...getOverrideProps(overrides, \\"Box\\")}>
      <CustomButton
        color=\\"#ff0000\\"
        width=\\"20px\\"
        buttonText=\\"Click Me\\"
        {...findChildOverrides(props.overrides, \\"CustomButton\\")}
      ></CustomButton>
    </View>
  );
}
"
`;

exports[`amplify render tests theme should render the theme 1`] = `
"/* eslint-disable */
import React from \\"react\\";
import { useEffect } from \\"react\\";
import {
  AmplifyProvider,
  DeepPartial,
  Theme,
  extendTheming,
} from \\"@aws-amplify/ui-react\\";
export default function withTheme<T>(
  WrappedComponent: React.ReactComponent<T>
) {
  const theme = {
    components: {
      alert: {
        backgroundColor: \\"hsl(210, 5%, 90%)\\",
        padding: \\"0.75rem 1rem\\",
        info: { backgroundColor: \\"hsl(220, 85%, 85%)\\" },
        error: { backgroundColor: \\"hsl(0, 75%, 85%)\\" },
        warning: { backgroundColor: \\"hsl(30, 75%, 85%)\\" },
        success: { backgroundColor: \\"hsl(130, 75%, 85%)\\" },
      },
    },
    borderWidths: { small: \\"1px\\", medium: \\"2px\\", large: \\"3px\\" },
    breakpoints: {
      values: { base: 0, small: 30, medium: 48, large: 62, xl: 80, xxl: 96 },
      unit: \\"em\\",
      defaultBreakpoint: \\"base\\",
    },
    testAllTypes: {
      null: null,
      string: \\"test\\",
      number: 10,
      boolean: { t: true, false: false },
      object: { object: { test: \\"test\\" }, array: [\\"test\\"] },
    },
  };
  const displayName =
    WrappedComponent.displayName || WrappedComponent.name || \\"Component\\";
  const ComponentWithTheme = (props: T) => {
    const theming = extendTheming(theme);
    useEffect(() => {
      Object.entries(theming.CSSVariables).forEach(([key, value]) => {
        document.documentElement.style.setProperty(key, value);
      });
    });
    return (
      <AmplifyProvider theming={theming}>
        <WrappedComponent {...props} />
      </AmplifyProvider>
    );
  };
  ComponentWithTheme.displayName = displayName;
  return ComponentWithTheme;
}
"
`;

exports[`amplify render tests theme should render the theme with ES5 1`] = `
"var __assign =
  (this && this.__assign) ||
  function () {
    __assign =
      Object.assign ||
      function (t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
          s = arguments[i];
          for (var p in s)
            if (Object.prototype.hasOwnProperty.call(s, p)) t[p] = s[p];
        }
        return t;
      };
    return __assign.apply(this, arguments);
  };
/* eslint-disable */
import React from \\"react\\";
import { useEffect } from \\"react\\";
import { AmplifyProvider, extendTheming } from \\"@aws-amplify/ui-react\\";
export default function withTheme(WrappedComponent) {
  var theme = {
    components: {
      alert: {
        backgroundColor: \\"hsl(210, 5%, 90%)\\",
        padding: \\"0.75rem 1rem\\",
        info: { backgroundColor: \\"hsl(220, 85%, 85%)\\" },
        error: { backgroundColor: \\"hsl(0, 75%, 85%)\\" },
        warning: { backgroundColor: \\"hsl(30, 75%, 85%)\\" },
        success: { backgroundColor: \\"hsl(130, 75%, 85%)\\" },
      },
    },
    borderWidths: { small: \\"1px\\", medium: \\"2px\\", large: \\"3px\\" },
    breakpoints: {
      values: { base: 0, small: 30, medium: 48, large: 62, xl: 80, xxl: 96 },
      unit: \\"em\\",
      defaultBreakpoint: \\"base\\",
    },
    testAllTypes: {
      null: null,
      string: \\"test\\",
      number: 10,
      boolean: { t: true, false: false },
      object: { object: { test: \\"test\\" }, array: [\\"test\\"] },
    },
  };
  var displayName =
    WrappedComponent.displayName || WrappedComponent.name || \\"Component\\";
  var ComponentWithTheme = function (props) {
    var theming = extendTheming(theme);
    useEffect(function () {
      Object.entries(theming.CSSVariables).forEach(function (_a) {
        var key = _a[0],
          value = _a[1];
        document.documentElement.style.setProperty(key, value);
      });
    });
    return React.createElement(
      AmplifyProvider,
      { theming: theming },
      React.createElement(WrappedComponent, __assign({}, props))
    );
  };
  ComponentWithTheme.displayName = displayName;
  return ComponentWithTheme;
}
"
`;

exports[`amplify render tests theme should render the theme with TSX 1`] = `
"/* eslint-disable */
import React from \\"react\\";
import { useEffect } from \\"react\\";
import {
  AmplifyProvider,
  DeepPartial,
  Theme,
  extendTheming,
} from \\"@aws-amplify/ui-react\\";
export default function withTheme<T>(
  WrappedComponent: React.ReactComponent<T>
) {
  const theme = {
    components: {
      alert: {
        backgroundColor: \\"hsl(210, 5%, 90%)\\",
        padding: \\"0.75rem 1rem\\",
        info: { backgroundColor: \\"hsl(220, 85%, 85%)\\" },
        error: { backgroundColor: \\"hsl(0, 75%, 85%)\\" },
        warning: { backgroundColor: \\"hsl(30, 75%, 85%)\\" },
        success: { backgroundColor: \\"hsl(130, 75%, 85%)\\" },
      },
    },
    borderWidths: { small: \\"1px\\", medium: \\"2px\\", large: \\"3px\\" },
    breakpoints: {
      values: { base: 0, small: 30, medium: 48, large: 62, xl: 80, xxl: 96 },
      unit: \\"em\\",
      defaultBreakpoint: \\"base\\",
    },
    testAllTypes: {
      null: null,
      string: \\"test\\",
      number: 10,
      boolean: { t: true, false: false },
      object: { object: { test: \\"test\\" }, array: [\\"test\\"] },
    },
  };
  const displayName =
    WrappedComponent.displayName || WrappedComponent.name || \\"Component\\";
  const ComponentWithTheme = (props: T) => {
    const theming = extendTheming(theme);
    useEffect(() => {
      Object.entries(theming.CSSVariables).forEach(([key, value]) => {
        document.documentElement.style.setProperty(key, value);
      });
    });
    return (
      <AmplifyProvider theming={theming}>
        <WrappedComponent {...props} />
      </AmplifyProvider>
    );
  };
  ComponentWithTheme.displayName = displayName;
  return ComponentWithTheme;
}
"
`;

exports[`amplify render tests user specific attributes should render user specific attributes 1`] = `
"/* eslint-disable */
import React from \\"react\\";
import {
  Button,
  EscapeHatchProps,
  Flex,
  Image,
  getOverrideProps,
} from \\"@aws-amplify/ui-react\\";

export type ProfileProps = {
  overrides?: EscapeHatchProps | undefined | null,
};
export default function Profile(props: ProfileProps): JSX.Element {
  const {} = props;
  const overrides = { ...props.overrides };
  const {
    attributes: {
      username,
      picture: userImage,
      [\\"custom:favorite_icecream\\"]: customUserAttributeIcecream,
    },
  } = useAuthenticatedUser();
  return (
    <Flex {...props} {...getOverrideProps(overrides, \\"Flex\\")}>
      <Image
        src={userImage}
        {...getOverrideProps(overrides, \\"Flex.Image\\")}
      ></Image>
      <Button
        label={username}
        {...getOverrideProps(overrides, \\"Flex.Button\\")}
      ></Button>
      <Button
        label={customUserAttributeIcecream}
        {...getOverrideProps(overrides, \\"Flex.Button\\")}
      ></Button>
    </Flex>
  );
}
"
`;
